File: Python algo/Arrays/2-maxProfit.py

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        i = 0
        j = 0
        print(prices[:len(prices) - 1])
        for i, _ in enumerate(prices[:len(prices) - 1]):
            if prices[i] > prices[i+1]:
                profit += prices[i] - prices[j]
                j = i + 1
        
        profit += prices[len(prices) - 1] - prices[j]

        return profit

sol = Solution()
print(sol.maxProfit([1,2,3,4,5]))

File: js algo/arrays/2-maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var profit = 0
    var j = 0
    for (var i=0; i < prices.length - 1; i++) {
        if (prices[i] > prices[i+1]) {
            profit += prices[i] - prices[j]
            j = i + 1
        }
    }
    
    profit += prices[prices.length - 1] - prices[j]

    return profit
};

console.log(maxProfit([1,2,3,4,5]))